src/environment.js

```diff
@@ -104,6 +112,8 @@ export type Binding = {
   environment: EnvironmentRecord,
   name: string,
   isGlobal: boolean,
+  isTainted: boolean,
```

Comment:
How about `hasLeaked` to stay with the leaking terminology?

Reply:
Yea, in this case that seems pretty straightforward. Although this suffers from the same problem as the object flag. It really should vary between branches and get joined.